rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/data/TaskResultImpl.java

```diff
@@ -71,7 +82,17 @@ public FlowAction getAction() {
 
     @Override
     public Throwable getException() {
-        throw new UnsupportedOperationException();
+        if (serializedException == null || exceptionMessage == null) {
+            return null;
+        }
+        try {
+            Throwable unserializedException = (Throwable) object(serializedException);
+            return unserializedException;
+        } catch (ClassCastException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1166)